### ISSUE TITLE
Update of versions for dependencies

### ISSUE THREAD

**Issue by reitzmichnicht** [qoomon/maven-git-versioning-extension#65]
I have a last issue that could be improved by this great extension.
Please see this demo project:
https://github.com/reitzmichnicht/versioning-demo

It is using the version of modules inside its own dependency section.
The maven-versions-plugin or the maven-release-plugin also change them if the version used there is identical to the old version. Without a fix the build breaks if the old versions are not already installed to the repo :(

Perhaps you can have a look at their code
https://github.com/mojohaus/versions-maven-plugin/blob/master/src/main/java/org/codehaus/mojo/versions/api/PomHelper.java

**Comment by qoomon** [comment]
You can make use of `${project.version}` instead of hardcoded version. Would that solve your problem?

e.g.
```xml
<dependency>
        <groupId>com.foo</groupId>
        <artifactId>cit-it-depchain</artifactId>
        <version>${project.version}</version>
        <type>pom</type>
</dependency>
```

**Comment by qoomon** [comment]
Otherwise I need to process all dependency sections, and there are a lot of possible places.
dependencies, dependency management, plugins, plugin management,...

**Comment by qoomon** [comment]
I think it would be nice to have this feature anyway, so I'll try to implement this within the next week.

**Comment by reitzmichnicht** [comment]
Sadly project.version is replaced by the version of the actual pom, not the version of the pom where it is defined. So if I use my demo pom as parent for another project C the dependencies try to resolve with C.version.

**Comment by qoomon** [comment]
Thats why => https://stackoverflow.com/a/16895876/5376635

**Comment by qoomon** [comment]
this is another way to solve it https://maven.apache.org/maven-ci-friendly.html
however I'try to implement the update of dependencies as well.

**Comment by reitzmichnicht** [comment]
Using the revision property also works great, but then again the versioning extension does not handle it correctly. I would prefer support for the revision property instead of scanning dependencies to be honest.

**Comment by qoomon** [comment]
Any suggestions how it should behave? I am a little lost how to implement this in a proper way.

**Comment by reitzmichnicht** [comment]
I'm not sure how this is handled internally, but in case you find `<version>${revision}</version>` in a pom, your extension should just update the revision property, maven handles everything else.

**Comment by qoomon** [comment]
What about `<version>${revision}-something</version>`?

**Comment by reitzmichnicht** [comment]
Yes I was also thinking about that. Perhaps you should log a warning that version replacement might not work as expected. I tested it with the current version of your extension and in general it works until someone uses the variables inside the version tag also in some other places.

**Comment by qoomon** [comment]
@reitzmichnicht I haven't had time to work on that, however you might be able to utilize the parameter replacements e.g.
```xml
...
<branch>
        <pattern>master</pattern>
        <versionFormat>${revision}</versionFormat>
        <property>
              <pattern>revision</pattern>
              <value>
                     <format>${branch}-SNAPSHOT</format>
              </value>
        </property>
</branch>
...
```

**Comment by reitzmichnicht** [comment]
I did some new tests and sadly this whole revision property thing does not work as expected. I missed one sentence in chapter install http://maven.apache.org/maven-ci-friendly.html:

> If you like to install or deploy artifacts by using the above setup you have to use the flatten-maven-plugin otherwise you will install/deploy artifacts in your repository which will not be consumable by Maven anymore

I had only a look at the effective pom, not the one that was installed in the repo.

So if someone uses revision, your update parameter feature should be sufficient, but I think not everyone wants to run the flatten plugin to get the feature branch stuff working.

**Comment by reitzmichnicht** [comment]
Together with the flatten plugin updating the revision property works as expected, but the flatten plugin has some nasty side effects as it modifies more of the pom than it should.

**Comment by qoomon** [comment]
That's right. So then I'll try to do the update properties and dependencies if the have an exact version match.

**Comment by qoomon** [comment]
It's implemented in in version `6.0.4`
